flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/metadata/FlinkRelMdUniqueKeys.scala

```diff
@@ -244,7 +243,25 @@ class FlinkRelMdUniqueKeys private extends MetadataHandler[BuiltInMetadata.Uniqu
       ignoreNulls: Boolean): JSet[ImmutableBitSet] = {
     val inputUniqueKeys = mq.getUniqueKeys(rel.getInput, ignoreNulls)
     val rankFunColumnIndex = RankUtil.getRankNumberColumnIndex(rel).getOrElse(-1)
-    if (rankFunColumnIndex < 0) {
+    //TODO current deduplicate on row time is still a Rank,
+    // remove this after support deduplicate on row time
+    val canConvertToDeduplicate: Boolean = {
```

Comment:
it's better we can put them into a separate commit

Reply:
yes, this part will be rework in the commit of supporting deduplicate on row time.